zeeguu/core/model/user.py

```diff
@@ -201,21 +201,35 @@ def has_bookmarks(self):
 
     def bookmarks_to_study(self, bookmark_count=10):
         """
+        We now use a logic to sort the words, if we call this everytime
```

Comment:
Uff. It's hard for me to make sense of the difference between these two. 
What I wonder now is the following:
- the similar words is only used in MultipleChoice.js

Why don't we ask for more words at the beginning of the exercise session, and just pass them to the exercise at creation time. Then we don't have to reach out from within the exercise to get "similar words". 

We can then completely remove the similar words function and endpoint, which is anyway not a great implementation...

Reply:
That could be a way to go. 

I also did work on the similar words in my own project using those confusion word sets and the NLP similarity, so maybe at some point we can just move the similar word call to query that rather than the current logic. However, I think it could be nice to have a way of `get_user_words(n)` to get some words the user has seen in the system. 

The reason why I left it in the system is because I believe it might still be useful as a way to "distract" users but ensure it's words they have seen and not something random.

So maybe we should just rename the end point? What do you think?

Reply:
You're right. Distraction words that come from the set of translated words still makes sense. How would you rename the endpoint? Something like: `get_random_bookmarks`? Could take as input a set of bookmark ids to be excluded, so we don't by chance stumble upon words that are being studied at the moment?

Reply:
Mhmm, the thing is that it's not random, it's from the scheduled_list, maybe just calling filtered_scheduled_bookmarks(id_set) would make the most sense, so we could pass it the ids of the scheduled words and avoid collisions, and if not possible we resort to similar words?

Reply:
It could, but again we are sorting the objects every time so I don't think it would be smart to do that. Instead, we would have a more "relaxed" query that just excludes the ids and doesn't need to sort them or filter them by duplicates.

Reply:
you could take one extra parameter for sorting? :) 
that being said, that method would become more complicated maybe? in that case let's go with your idea!